dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubHelper.java

```diff
@@ -42,10 +47,50 @@
 public final class GitHubHelper {
 
     private static final Logger LOG = LoggerFactory.getLogger(GitHubHelper.class);
+    public static final String BRANCHNAME_FOR_BOT = "feature/add_dockstore_yml";
+
+    public static final String DOCKSTORE_BOT_PR_TEXT = """
+           Dockstore's bot has guessed at a .dockstore.yml for your repository.
+           
+           Please review, make appropriate changes, add any [additional keys](https://docs.dockstore.org/en/stable/assets/templates/template.html) and merge in order to complete integration with Dockstore, allowing Dockstore to keep informed of new changes to your workkflow, notebook, or tools.
+           """;
 
     private GitHubHelper() {
     }
 
+    /**
+     * Experimental building block, given a repository, create a fork of it and a PR to it
+     * @param inferredDockstoreYml should be the only real change in the PR
+     * @param gitHub    The GitHub API
+     * @param repositoryName    Name of the GitHub repository (e.g. dockstore/lambda)
+     * @param targetBranch    Name of the branch in the GitHub repository to make a PR against (e.g. develop)
+     * @return TBD
+     */
+    public static URL createForkPlusPR(String inferredDockstoreYml, GitHub gitHub, String repositoryName, String targetBranch) throws ApiException {
+        try {
+            String username = gitHub.getMyself().getLogin();
+            GHRepository targetRepository = gitHub.getRepository(repositoryName);
+            // note: this will try to turn an async call to a sync call by waiting up to half a minute, 3 seconds at a time.
+            // This did not seem consistently reliable, may need to wait more before creating a ref/branch
+            GHRepository fork = targetRepository.fork();
+            Thread.sleep(Duration.ofMinutes(1L).toMillis());
```

Comment:
Could we poll (check every 10 seconds or similar) to see if the fork was created?

Reply:
Unclear, the conversation in the library seems to be uncertain whether it is a wait in the first place
https://github.com/hub4j/github-api/pull/1095#issuecomment-819633466